github/resource_github_branch.go

```diff
@@ -0,0 +1,202 @@
+
+	log.Printf("[DEBUG] Creating GitHub branch reference %s/%s (%s)",
+		orgName, repoName, branchRefName)
+	_, _, err = client.Git.CreateRef(ctx, orgName, repoName, &github.Reference{
+		Ref:    &branchRefName,
+		Object: &github.GitObject{SHA: &sourceBranchSHA},
+	})
+	if err != nil {
+		return fmt.Errorf("Error creating GitHub branch reference %s/%s (%s): %s",
+			orgName, repoName, branchRefName, err)
+	}
+
+	d.SetId(buildTwoPartID(&repoName, &branchName))
+
+	return resourceGithubBranchRead(d, meta)
+}
+
+func resourceGithubBranchRead(d *schema.ResourceData, meta interface{}) error {
+	err := checkOrganization(meta)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+	if !d.IsNewResource() {
+		ctx = context.WithValue(ctx, ctxEtag, d.Get("etag").(string))
+	}
+
+	client := meta.(*Organization).client
+	orgName := meta.(*Organization).name
+	repoName, branchName, err := parseTwoPartID(d.Id(), "repository", "branch")
+	if err != nil {
+		return err
+	}
+	branchRefName := "refs/heads/" + branchName
+
+	log.Printf("[DEBUG] Querying GitHub branch reference %s/%s (%s)",
+		orgName, repoName, branchRefName)
+	ref, resp, err := client.Git.GetRef(ctx, orgName, repoName, branchRefName)
+	if err != nil {
+		if ghErr, ok := err.(*github.ErrorResponse); ok {
+			if ghErr.Response.StatusCode == http.StatusNotModified {
+				return nil
+			}
+			if ghErr.Response.StatusCode == http.StatusNotFound {
+				log.Printf("[WARN] Removing branch %s/%s (%s) from state because it no longer exists in Github",
+					orgName, repoName, branchName)
+				d.SetId("")
+				return nil
+			}
+		}
+		return fmt.Errorf("Error querying GitHub branch reference %s/%s (%s): %s",
+			orgName, repoName, branchRefName, err)
+	}
+
+	d.SetId(buildTwoPartID(&repoName, &branchName))
+	d.Set("etag", resp.Header.Get("ETag"))
+	d.Set("repository", repoName)
+	d.Set("branch", branchName)
+	d.Set("ref", *ref.Ref)
+	d.Set("sha", *ref.Object.SHA)
+
+	return nil
+}
+
+func resourceGithubBranchDelete(d *schema.ResourceData, meta interface{}) error {
+	err := checkOrganization(meta)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+
+	client := meta.(*Organization).client
+	orgName := meta.(*Organization).name
+	repoName, branchName, err := parseTwoPartID(d.Id(), "repository", "branch")
+	if err != nil {
+		return err
+	}
+	branchRefName := "refs/heads/" + branchName
+
+	log.Printf("[DEBUG] Deleting GitHub branch reference %s/%s (%s)",
+		orgName, repoName, branchRefName)
+	_, err = client.Git.DeleteRef(ctx, orgName, repoName, branchRefName)
+	if err != nil {
+		return fmt.Errorf("Error deleting GitHub branch reference %s/%s (%s): %s",
+			orgName, repoName, branchRefName, err)
+	}
+
+	return nil
+}
+
+func resourceGithubBranchImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	repoName, branchName, err := parseTwoPartID(d.Id(), "repository", "branch")
```

Comment:
```suggestion
	_, branchName, err := parseTwoPartID(d.Id(), "repository", "branch")
```

`repoName` is not needed as you don't need to set the ID [here](https://github.com/terraform-providers/terraform-provider-github/pull/364/files#diff-81bb68b8177b7f989f656af57eebc7afR196).